pandas/core/frame.py

```diff
@@ -3966,6 +3966,24 @@ def __setitem__(self, key, value):
         ):
             # Column to set is duplicated
             self._setitem_array([key], value)
+        elif (
+            isinstance(value, np.ndarray) and value.ndim > 1 and self.columns.is_unique
+        ):
+            # TODO: a check for MultiIndex should be added
```

Comment:
does the check on the next line not take care of this TODO?

Reply:
> does the check on the next line not take care of this TODO?

No, the check here is just placeholder for further improvements, and the real check should be inserted between "isinstance" line and "_set_item" line. I did not figure out what and how checks for MultiIndex should be added so I skipped this. But I think the current PR is a mitigation covering and fixing frequently used scenes.